File: ros_ws/src/my_robot_tutorials/src/number_publisher.cpp
#include <ros/ros.h>
#include <std_msgs/Int64.h>

int main(int argc, char **argv)
{
    ros::init(argc, argv, "number_publisher");
    ros::NodeHandle nh;
    ros::Publisher publisher = nh.advertise<std_msgs::Int64>("number", 10);
    double frequency;
    nh.getParam("/number_publisher_freq", frequency);
    ros::Rate rate(frequency);
    const int number = 4;
    ROS_INFO("number_publisher node has been created");
    
    while(ros::ok()) {
        std_msgs::Int64 msg;
        msg.data = number;
        publisher.publish(msg);
        rate.sleep();
    }

    return 0;
}

File: ros_ws/src/my_robot_tutorials/scripts/number_publisher.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import Int64


def main(args=None):
    rospy.init_node("number_publisher")

    number = 4
    publisher = rospy.Publisher("number", Int64, queue_size=10)
    rate = rospy.Rate(1.0 / 2.0)

    rospy.loginfo("number_publisher node has been created")

    while not rospy.is_shutdown():
        msg = Int64()
        msg.data = number
        publisher.publish(msg)
        rate.sleep()


if __name__ == "__main__":
    main()